src/ws/cockpitauth.c

```diff
@@ -892,13 +895,20 @@ on_transport_closed (CockpitTransport *transport,
   else if (!session->initialized)
     {
       pipe = cockpit_pipe_transport_get_pipe (COCKPIT_PIPE_TRANSPORT (transport));
+      g_autofree gchar *captured_error = cockpit_pipe_take_stderr_as_utf8 (pipe);
+
       if (cockpit_pipe_get_pid (pipe, NULL))
         status = cockpit_pipe_exit_status (pipe);
       g_debug ("%s: authentication process exited: %d; problem %s", session->name, status, problem);
 
+      if (captured_error)
+        {
+          g_set_error (&error, COCKPIT_ERROR, COCKPIT_ERROR_AUTHENTICATION_FAILED,
+                       "captured-stderr:%s", captured_error);
+        }
       /* we get "access-denied" both if cockpit-session cannot execute cockpit-bridge (common case)
        * and if cockpit-session itself is not executable (corner case, messed up install) */
-      if (problem && (!session->authorize || g_strcmp0 (problem, "access-denied") != 0))
+      else if (problem && (!session->authorize || g_strcmp0 (problem, "access-denied") != 0))
         {
           g_set_error (&error, COCKPIT_ERROR, COCKPIT_ERROR_FAILED,
                        g_strcmp0 (problem, "no-cockpit") == 0
```

Comment:
IMHO it should first look at known error codes from the protocol (such as `no-cockpit`), and *then* fall back to showing raw stderr. I.e. the ":" branch here with "internal error" should show stderr (if present), and this could even happen unconditionally.